### ISSUE TITLE
Trouble clearing doctrine cache

### ISSUE THREAD

**Issue by PaddyLock** [Atlantic18/DoctrineExtensions#1621]
I am having trouble clearing the doctrine cache when using Translatable. When I use 

php app/console cache:flush doctrine --env=prod

The doctrine cache is cleared and new data is loaded correctly the first time the page is loaded. But then from the second page load the it reverts to the old cache. The result set being loaded is using Translatable Tree Walker as below. The strange thing is that it woks fine on my dev and staging environment. The problem only exists on production.

`public function findOneBySlugTranslatableWithSubCategories($slug, $locale)
    {
        $qb = $this->createQueryBuilder('c');

        $query = $qb->select('c', 'sc')
            ->leftJoin('c.subCategories', 'sc')
            ->where('c.slug = :slug')
            ->setParameter('slug', $slug)
            ->getQuery();

        // set to false to ensure left join loads translations
        $query->setHint(Query::HINT_FORCE_PARTIAL_LOAD, false);

        // set the translation query hint
        $query->setHint(
            \Doctrine\ORM\Query::HINT_CUSTOM_OUTPUT_WALKER,
            'Gedmo\\Translatable\\Query\\TreeWalker\\TranslationWalker'
        );

        // locale
        $query->setHint(
            \Gedmo\Translatable\TranslatableListener::HINT_TRANSLATABLE_LOCALE,
            $locale // take locale from session or request etc.
        );
        // fallback
        $query->setHint(
            \Gedmo\Translatable\TranslatableListener::HINT_FALLBACK,
            1 // fallback to default values in case if record is not translated
        );

        try {
            return $query->useResultCache(true,86400)->getSingleResult();
        } catch (NoResultException $e) {
            return null;
        }
    }`

**Comment by l3pp4rd** [comment]
hi, are you sure it flushes all doctrine caches? there is metadata, query, and result caches - 3 types of cache in doctrine. check your cache after clearing if there are no entries remaining. translatable was tested and proved many times to work correctly with cashing, it is most probably related to your caching environment

**Comment by PaddyLock** [comment]
Yes I cleared all the caches with

```
app/console doctrine:cache:clear-metadata --env=prod
app/console doctrine:cache:clear-query --env=prod
app/console doctrine:cache:clear-result --env=prod

```

I have now found the real problem. I was rendering an embedded controller, which itself made a doctrine call to load some entities. The entities which were loaded where stored in some internal cache and then reused later in the application, even though I was making a fresh doctrine call. To solve this I had to call `$entityManager->clear()` before subsequent queries, to force reload of entities from database. However, weirdly this only seems to work on my production environment, and has no effect on my dev environment.
